builder.go

```diff
@@ -31,6 +32,14 @@ type Copy struct {
 	// to the executor for handling.
 	Chown string
 	Chmod string
+	IsHeredoc bool
+	Files   []File
+}
+
+type File struct {
+	Command string
+	Name    string
+	Data    string
```

Comment:
These need some godoc to indicate to a caller what they're supposed to do with these new fields.